packages/terra-clinical-result/src/flowsheetresultcell/FlowsheetResultCell.jsx

```diff
@@ -0,0 +1,277 @@
+import ClinicalResult from '../ClinicalResult';
+import ClinicalResultBloodPressure from '../ClinicalResultBloodPressure';
+import observationPropShape from '../proptypes/observationPropTypes';
+import EnteredInError from '../common/other/_EnteredInError';
+import ResultError from '../common/other/_ResultError';
+import NoData from '../common/other/_KnownNoData';
+import NumericOverflow from '../common/other/_NumericOverflow';
+import { isEmpty, checkIsStatusInError, checkIsOfType } from '../common/utils';
+import styles from './FlowsheetResultCell.module.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   *  A set of clinical results.                                                          .
+   */
+  resultDataSet: PropTypes.arrayOf(PropTypes.shape({
+    /**
+     *  A single clinical result or blood pressure result id.
+     */
+    id: PropTypes.string,
+    /**
+     *  A single clinical result or blood pressure result.
+     */
+    resultData: observationPropShape,
+  })),
+  /**
+   * Visually hides the unit of measure when presented in a series of side-by-side columns of the same unit.
+   */
+  hideUnit: PropTypes.bool,
+  /**
+   * The padding styling to apply to the Time Column Header Cell.
+   * One of `'none'`, `'standard'`, `'compact'`.
+   */
+  paddingStyle: PropTypes.oneOf(['none', 'standard', 'compact']),
+  /**
+   * Override that shows an Error display. Used when there is a known error or problem when retrieving or assembling the clinical result data.
+   */
+  hasResultError: PropTypes.bool,
+  /**
+   * Override that shows a known "No Data" display. Used when there is known to be no value for a given clinical result concept at a specific datetime.
+   */
+  hasResultNoData: PropTypes.bool,
+  /**
+   * @private
+   * The intl object to be injected for translations.
+   */
+  intl: intlShape.isRequired,
+};
+
+const defaultProps = {
+  resultDataSet: [],
+  hideUnit: false,
+  paddingStyle: 'compact',
+  hasResultError: false,
+  hasResultNoData: false,
+};
+
+const FlowsheetResultCell = (props) => {
+  const {
+    resultDataSet,
+    hideUnit,
+    paddingStyle,
+    hasResultError,
+    hasResultNoData,
+    intl,
+    ...customProps
+  } = props;
+  const containerDiv = useRef(null);
+  const [contentWidth, setContentWidth] = useState(null);
+  const [numericOverflow, setNumericOverflow] = useState(false);
+
+  useEffect(() => {
+    if (containerDiv.current && resultDataSet[0]) {
+      if (checkIsOfType(resultDataSet[0], 'NUMERIC')) {
+        if (!contentWidth) {
+          setContentWidth(containerDiv.current.children[0].getBoundingClientRect().width);
+        }
+        if (containerDiv.current.getBoundingClientRect().width <= contentWidth && !numericOverflow) {
+          setNumericOverflow(true);
+        } else if (containerDiv.current.getBoundingClientRect().width > contentWidth) {
+          setNumericOverflow(false);
+        }
+      }
+    }
```

Comment:
can we move the first conditional to a guard clause to reduce the number of nested conditionals? i also think we should store the container's width inside a local variable to avoid accessing that property twice. this will also improve readability
```suggestion
    if (!containerDiv.current || !resultDataSet[0]) {
      return;
    }
    
    if (checkIsOfType(resultDataSet[0], 'NUMERIC')) {
        if (!contentWidth) {
          setContentWidth(containerDiv.current.children[0].getBoundingClientRect().width);
        }
        let containerWidth = containerDiv.current.getBoundingClientRect().width;
        if (containerWidth <= contentWidth && !numericOverflow) {
          setNumericOverflow(true);
        } else if (containerWidth > contentWidth) {
          setNumericOverflow(false);
        }
      }
```

Reply:
Applied here https://github.com/cerner/terra-clinical/pull/574/commits/4015e0f2acf433faa34429946f05e3db8a2ebebc